editor/src/components/inspector/sections/component-section/variables-in-scope-utils.ts

```diff
@@ -395,7 +398,11 @@ export function useVariablesInScopeForSelectedElement(
       // if the selected element doesn't have an associacted variablesInScope, we assume it's safe to walk up the path within the same component
       let path = selectedViewPath
       while (
-        EP.pathsEqual(EP.getContainingComponent(path), EP.getContainingComponent(selectedViewPath))
+        EP.pathsEqual(
```

Comment:
I'm like 90% sure we used to have a function which would create all of the ancestor paths of a given element path. Maybe we deleted that at some point, but that sounds much safer than using a `while` loop here

Reply:
what about using `EP.getAncestors` (or possibly `findAmongAncestorsOfPath`) and filtering over those results?

Reply:
fixed in ade133944!